src/object/srv_obj_migrate.c

```diff
@@ -1261,6 +1339,68 @@ migrate_obj_iter_cb(daos_handle_t ih, d_iov_t *key_iov, d_iov_t *val_iov,
 	return rc;
 }
 
+/* Destroys a container exactly one time per migration session. Uses the
+ * mpt_cont_dest_tab field of the tls to store which containers have already
+ * been deleted this session.
+ *
+ * Only used for reintegration
+ */
+static int
+destroy_existing_container(struct migrate_pool_tls *tls, uuid_t cont_uuid)
+{
+	d_list_t *link;
+	int rc;
+
+	link = d_hash_rec_find(&tls->mpt_cont_dest_tab, cont_uuid,
+			       sizeof(uuid_t));
```

Comment:
we should check the hash table, destroy container and recreate container in migrate_one_ult() instead of doing collective destroy?
Collective destroy can race with other ULTs (migrate ULT)

Reply:
Di came up with a pretty clean solution for this. Not sure what to do if there's a data race here - this is here at Di's recommendation.

Reply:
Liang: what race do you mean here? I may miss sth.
here is where the container is being accessed(opened) firstly during migration, migrate_one_ult would be too late for this.